lxd/response/smart.go

```diff
@@ -23,12 +25,25 @@ func SmartError(err error) Response {
 
 	statusCode, found := api.StatusErrorMatch(err)
 	if found {
+		// If the status code relates to an authentication error, return the generic HTTP status text.
+		// This is so that there is no difference in the error message when
+		// - a resource really doesn't exist (e.g. "Instance not found").
```

Comment:
Ah so this isn't going to work, because as far as I can tell, we are now always converting "instance not found" type errors into generic "not found" errors. Which undoes all the work we've done over the last few years to pass more intelligible errors back to the end user - i.e if you are doing something on an instance that depends on a profile and the profile doesn't exist you get a 404 "profile not found" rather than a 404 "not found" and you can't tell if its the instance or the profile that is not found.

This would be a retrograde step.

Reply:
Sure thing. We only need to mask errors in the pre-processing step before the call to the authorizer. So I can revert back to the previous method.

Reply:
Done. Will re-request a review once the tests pass.